src/main/java/org/jabref/gui/exporter/SaveGitDatabaseAction.java

```diff
@@ -0,0 +1,41 @@
+package org.jabref.gui.exporter;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import org.jabref.logic.git.GitHandler;
+import org.jabref.preferences.PreferencesService;
+
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SaveGitDatabaseAction {
+    static final Logger LOGGER = LoggerFactory.getLogger(GitHandler.class);
+    final Path filePath;
+    final String automaticCommitMsg = "Automatic update via JabRef";
+    private final PreferencesService preferences;
+
+    public SaveGitDatabaseAction(Path filePath, PreferencesService preferences) {
+        this.filePath = filePath;
+        this.preferences = preferences;
+    }
+
+    /**
+    * Handle JabRef git integration action
+    *
+    * @return true of false whether the action was successful or not
+    */
+    public boolean automaticUpdate() {
+        try {
+            GitHandler git = new GitHandler(this.filePath.getParent());
+            git.setGitPreferences(preferences.getGitPreferences());
+            git.createCommitWithSingleFileOnCurrentBranch(this.filePath.getFileName().toString(), automaticCommitMsg, false);
```

Comment:
Boolen parameters are always an indicator for bad design. Either an ENUM should be used - or two different methods.

Here, you can just remove the parameter, because you always pass `false`.